lib/burnside/ast-builder.cc

```diff
@@ -602,6 +602,166 @@ inline void annotateFuncCFG(AST::FunctionLikeUnit &flu) {
+          [](const Pa::CompilerDirective *) { return "CompilerDirective"; },
+          [](const Pa::CriticalConstruct *) { return "CriticalConstruct"; },
+          [](const Pa::DoConstruct *) { return "DoConstruct"; },
+          [](const Pa::ForallConstruct *) { return "ForallConstruct"; },
+          [](const Pa::IfConstruct *) { return "IfConstruct"; },
+          [](const Pa::OmpEndLoopDirective *) { return "OmpEndLoopDirective"; },
+          [](const Pa::OpenMPConstruct *) { return "OpenMPConstruct"; },
+          [](const Pa::SelectRankConstruct *) { return "SelectRankConstruct"; },
+          [](const Pa::SelectTypeConstruct *) { return "SelectTypeConstruct"; },
+          [](const Pa::WhereConstruct *) { return "WhereConstruct"; },
+
+          [](const Pa::AssociateStmt *) { return "AssociateStmt"; },
+          [](const Pa::BlockStmt *) { return "BlockStmt"; },
+          [](const Pa::CaseStmt *) { return "CaseStmt"; },
+          [](const Pa::ChangeTeamStmt *) { return "ChangeTeamStmt"; },
+          [](const Pa::CriticalStmt *) { return "CriticalStmt"; },
+          [](const Pa::ElseIfStmt *x) { return "ElseIfStmt"; },
+          [](const Pa::ElseStmt *) { return "ElseStmt"; },
+          [](const Pa::ElsewhereStmt *) { return "ElsewhereStmt"; },
+          [](const Pa::EndAssociateStmt *) { return "EndAssociateStmt"; },
+          [](const Pa::EndBlockStmt *) { return "EndBlockStmt"; },
+          [](const Pa::EndChangeTeamStmt *) { return "EndChangeTeamStmt"; },
+          [](const Pa::EndCriticalStmt *) { return "EndCriticalStmt"; },
+          [](const Pa::EndDoStmt *) { return "EndDoStmt"; },
+          [](const Pa::EndForallStmt *) { return "EndForallStmt"; },
+          [](const Pa::EndIfStmt *) { return "EndIfStmt"; },
+          [](const Pa::EndSelectStmt *) { return "EndSelectStmt"; },
+          [](const Pa::EndWhereStmt *) { return "EndWhereStmt"; },
+          [](const Pa::ForallConstructStmt *) { return "ForallConstructStmt"; },
+          [](const Pa::IfThenStmt *x) { return "IfThenStmt"; },
+          [](const Pa::MaskedElsewhereStmt *) { return "MaskedElsewhereStmt"; },
+          [](const Pa::NonLabelDoStmt *) { return "NonLabelDoStmt"; },
+          [](const Pa::SelectCaseStmt *) { return "SelectCaseStmt"; },
+          [](const Pa::SelectRankCaseStmt *) { return "SelectRankCaseStmt"; },
+          [](const Pa::SelectRankStmt *) { return "SelectRankStmt"; },
+          [](const Pa::SelectTypeStmt *) { return "SelectTypeStmt"; },
+          [](const Pa::TypeGuardStmt *) { return "TypeGuardStmt"; },
+          [](const Pa::WhereConstructStmt *) { return "WhereConstructStmt"; },
+      },
+      e.u);
+}
+
+void dumpEvalList(
+    llvm::raw_ostream &o, std::list<AST::Evaluation> &evals, int indent = 1) {
+  static const std::string white{"                                      ++"};
+  std::string indentString{white.substr(0, indent * 2)};
```

Comment:
This actually has some nice properties.  Forced use of a negative value becomes a reasonable 0-indent; large indexes automatically top out at the length of the string with the "++" tail as intended.  Since this is dump code, I thought these exceptional cases worked reasonably.  Short and sweet.

Reply:
OK.